packages/js/product-editor/src/components/variations-table/variation-actions-menus/variation-actions.tsx

```diff
@@ -0,0 +1,150 @@
+/**
+ * External dependencies
+ */
+import { MenuGroup, MenuItem } from '@wordpress/components';
+import { createElement, Fragment } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { recordEvent } from '@woocommerce/tracks';
+import { ProductVariation } from '@woocommerce/data';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { VariationActionsMenuProps } from './types';
+import { TRACKS_SOURCE } from '../../../constants';
+import { ShippingMenuItem } from '../shipping-menu-item';
+import { InventoryMenuItem } from '../inventory-menu-item';
+import { PricingMenuItem } from '../pricing-menu-item';
+import { ToggleVisibilityMenuItem } from '../toggle-visibility-menu-item';
+import { DownloadsMenuItem } from '../downloads-menu-item';
+import { VariationActionsMenuItem } from './variation-actions-menu-item';
+import { QUICK_UPDATE, SINGLE_UPDATE } from './constants';
+import { UpdateStockMenuItem } from '../update-stock-menu-item';
+import { SetListPriceMenuItem } from '../set-list-price-menu-item';
+
+export function VariationActions( {
+	selection,
+	onChange,
+	onDelete,
+	onClose,
+	type = SINGLE_UPDATE,
+}: VariationActionsMenuProps & {
+	onClose: () => void;
+	type: string;
+} ) {
+	const isQuickUpdate = type === QUICK_UPDATE;
+	return (
+		<div
+			className={ classNames( {
+				'components-dropdown-menu__menu': isQuickUpdate,
+			} ) }
+		>
+			<MenuGroup
+				label={
+					isQuickUpdate
+						? undefined
+						: sprintf(
+								/** Translators: Variation ID */
+								__( 'Variation Id: %s', 'woocommerce' ),
+								( selection as ProductVariation ).id
+						  )
+				}
+			>
+				{ isQuickUpdate ? (
+					<>
+						<UpdateStockMenuItem
+							selection={ selection }
+							onChange={ onChange }
+							onClose={ onClose }
+						/>
+						<SetListPriceMenuItem
+							selection={ selection }
+							onChange={ onChange }
+							onClose={ onClose }
+						/>
+					</>
+				) : (
+					<MenuItem
+						href={ ( selection as ProductVariation ).permalink }
+						target="_blank"
+						rel="noreferrer"
+						onClick={ () => {
+							recordEvent( 'product_variations_preview', {
+								source: TRACKS_SOURCE,
+								variation_id: ( selection as ProductVariation )
+									.id,
+							} );
+						} }
+					>
+						{ __( 'Preview', 'woocommerce' ) }
+					</MenuItem>
+				) }
+
+				<ToggleVisibilityMenuItem
+					selection={ selection }
+					onChange={ onChange }
+					onClose={ onClose }
+				/>
+				<VariationActionsMenuItem.Slot
+					group={ '_main' }
```

Comment:
I'm not sure we need/want this slot.

Why the `_` in the front of the group name?